src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySource.cs

```diff
@@ -407,105 +407,94 @@ internal void NotifyActivityAddException(Activity activity, Exception exception,
         }
     }
 
-    // SynchronizedList<T> is a helper collection which ensure thread safety on the collection
-    // and allow enumerating the collection items and execute some action on the enumerated item and can detect any change in the collection
-    // during the enumeration which force restarting the enumeration again.
-    // Caution: We can have the action executed on the same item more than once which is ok in our scenarios.
+    //this class uses a copy-on-write design to ensure thread safety all operations are thread safe.
+    //However, it is possible for read-only operations to see stale versions of the item while a change
+    //is occurring. This should not be a practical issue if updates are infrequent
```

Comment:
```suggestion
    //is occurring.
```
In terms of consistency for the Activity scenario its fine even if the updates are frequent :)

Reply:
Fixed 😁